pkg/providers/shell/shell_provider.go

```diff
@@ -202,15 +201,16 @@ func (si *shellInstance) GetRoot() string {
 	return si.root
 }
 
-func (si *shellInstance) doDestroy(writer io.StringWriter, timeout time.Duration, err error) {
-	_, _ = writer.WriteString(fmt.Sprintf("Error during k8s API initialisation %v", err))
-	_, _ = writer.WriteString(fmt.Sprintf("Trying to destroy cluster"))
-	// In case we failed to start and create cluster utils.
-	err2 := si.Destroy(timeout)
-	if err2 != nil {
-		_, _ = writer.WriteString(fmt.Sprintf("Error during destroy of cluster %v", err2))
-	}
-}
+// TODO - this function was unused.  Should we delete it or should it be being used?
```

Comment:
This function was unused.  If its really unused, we should remove it.  If it should be used, let me know where and how :)

Reply:
Same here, not used :-) Thanks.